docs: Update CONTRIBUTING.md with Rust setup and turbo build info.

DevChthonic opened the pull request:
<!--
  Thank you for submitting a pull request!

  We appreciate the time and effort you have invested in making these changes. Please ensure that you provide enough information to allow others to review your pull request.

  Upon submission, your pull request will be automatically assigned with reviewers.

  If you want to learn more about contributing to this project, please visit: https://github.com/lynx-family/lynx-stack/blob/main/CONTRIBUTING.md.
-->

## Summary

<!-- Can you explain the reasoning behind implementing this change? What problem or issue does this pull request resolve? -->

This pull request updates `CONTRIBUTING.md` to address setup and build issues encountered by new contributors, resolving the problems documented in issue #710.

**Problem:** Following the previous instructions in `CONTRIBUTING.md` (`pnpm install` -> `pnpm build` -> `pnpm test`) leads to test failures, specifically the `Could not resolve "./dist/vitest.config"` error.

**Reasoning & Root Causes:**
1.  **Missing Prerequisite:** The project requires a specific Rust toolchain version, **`1.81.0`**, as defined in the `rust-toolchain` file at the root, along with the `wasm32-unknown-unknown` target for compiling native code (WASM/NAPI). This requirement was not documented, causing build failures during the Rust compilation steps for users who only had a different version (like the latest stable `1.86.0`) installed.
2.  **Incomplete Build Command:** The documented `pnpm build` command (which is an alias for `tsc --build`) only performs TypeScript compilation. It does *not* handle the necessary Rust compilation steps or other package-specific build scripts (like `rslib build` used by `@lynx-js/reactlynx-testing-library`) required to generate all artifacts needed for testing, including the missing `dist/vitest.config.js`.
3.  **Correct Build Process:** The full project build needs to be orchestrated by Turborepo using the command `pnpm exec turbo run build`. This ensures all dependencies (including Rust components) are built in the correct order before dependent packages attempt to use them.

**Solution Implemented in this PR:**
* The `CONTRIBUTING.md` file has been updated to:
    * Clearly list Rust `1.81.0` and the `wasm32-unknown-unknown` target as prerequisites, including the `rustup` command needed to install them.
    * Revise the "Building" section to emphasize that `pnpm exec turbo run build` is the required command for a complete build that must be run before `pnpm test`.
    * Clarify the limited scope of the root `pnpm build` (`tsc --build`) command.

These documentation changes will help future contributors set up their environment correctly and avoid the build/test failures initially reported.

<!-- It would be helpful if you could provide any relevant context, such as GitHub issues or related discussions. -->

close: https://github.com/lynx-family/lynx-stack/issues/710

## Checklist

<!--- Check and mark with an "x" -->

- [ ] Tests updated (or not required). **NOT REQUIRED**
- [x] Documentation updated (or not required). **CONTRIBUTING.md UPDATED**

cspell.jsonc: "rustup" added to **words**

Images 1: Cargo version `cargo --version` 1.81.0

![Screenshot 2025-05-08 at 11 31 10 AM](https://github.com/user-attachments/assets/de8af5ca-550c-4b54-af86-2be4ad7535d5)

Image 2: cspell.jsonc `"rustup" // Rust installer/manager`

![Screenshot 2025-05-08 at 11 29 49 AM](https://github.com/user-attachments/assets/048ce92d-ff10-4a9e-9b1d-f3e712ef9f4d)

Image 3: CONTRIBUTING.md updates

![Screenshot 2025-05-08 at 11 30 34 AM](https://github.com/user-attachments/assets/a5388f8c-1d09-4887-a8f5-7a58999657eb)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/lynx-family/lynx-stack?pullRequest=741) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/lynx-family/lynx-stack?pullRequest=741) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: DevChthonic<br/>:x: Chthonic<br/><hr/>**Chthonic** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/lynx-family/lynx-stack?pullRequest=741) it.</sub>

colinaaa commented on cspell.jsonc:
```diff
@@ -136,6 +136,7 @@
     "lhci",
     "lh",
     "queueMicrotask",
+    "rustup" // Rust installer/manager 
```
```suggestion
    "rustup", // Rust installer/manager 
```

formatting nitpick

DevChthonic replied:
Legendary catch

colinaaa commented on CONTRIBUTING.md:
```diff
@@ -18,31 +27,32 @@ pnpm install
 ### Building
 
-To build the projects, run:
+This project uses a combination of TypeScript, Rust, and custom build tools across multiple packages. A full build requires orchestration using Turborepo.
```
I would like to keep this part as simple as possible. Maybe just replace

- `pnpm build` -> `pnpm turbo build` 
- `pnpm build --watch` -> `pnpm turbo watch build`

We will review these documents once we have a more effective watching method.

DevChthonic replied:
On it, give me like 30 minutes.

colinaaa reviewed (commented):
Sorry for the late review. Thank you for your contribution!

We definitely need a setup guide for Rust, but it would be better not to specify version `1.81.0`, as it's easy to overlook updating it when changing the `rust-toolchain`.

For the building process, I prefer to keep it straightforward, similar to the testing or linting steps. Until we discover a more solid way, let's use `pnpm turbo build` for building and `pnpm turbo watch build` for watching changes.

DevChthonic commented:
Okay I've simplified the CONTRIBUTING.md file significantly based on your feedback @colinaaa. I've also gone ahead and tested `pnpm turbo build` and `pnpm turbo watch build` on my machine and they both work flawlessly.

I removed a comma from a `.ts` file and noted the terminal showing a problem. I then added the comma back and the terminal problem notification went away.

I have just committed the CONTRIBUTING.md file again. It's ready to review!

Images attached showcasing tests:

`pnpm turbo build`:
<img width="1787" alt="Screenshot 2025-05-13 at 6 03 58 AM" src="https://github.com/user-attachments/assets/d48026ef-9412-40d5-b221-90b1af2a4d84" />

`pnpm test`:
<img width="1789" alt="Screenshot 2025-05-13 at 6 04 36 AM" src="https://github.com/user-attachments/assets/3bdeac55-3ff7-4a9e-9b1c-036ada9e322d" />

`pnpm turbo watch build` (removed , > problem thrown):
<img width="1792" alt="Screenshot 2025-05-13 at 6 10 00 AM" src="https://github.com/user-attachments/assets/64391188-786e-4f3c-aedd-a0e141d1b21b" />

`pnpm turbo watch build` (added , > problem disappeared):
<img width="1791" alt="Screenshot 2025-05-13 at 6 10 11 AM" src="https://github.com/user-attachments/assets/f7a80a11-e375-42f7-abc8-70cab353d2eb" />

colinaaa reviewed (dismissed):
Thank you!

DevChthonic commented:
Hmmm, if I commit something that fixes the formatting will that cause this workflow to run again? Also what is the command for running `dprint`?

colinaaa commented:
> Hmmm, if I commit something that fixes the formatting will that cause this workflow to run again?

Yes, it will.

> Also what is the command for running `dprint`?

Run `pnpm dprint fmt`. Consider adding this command to the `CONTRIBUTING.md` file. It should ideally execute automatically via git hooks, though it's unclear why it isn't functioning for you.

DevChthonic commented:
> > Hmmm, if I commit something that fixes the formatting will that cause this workflow to run again?
> 
> Yes, it will.
> 
> > Also what is the command for running `dprint`?
> 
> Run `pnpm dprint fmt`. Consider adding this command to the `CONTRIBUTING.md` file. It should ideally execute automatically via git hooks, though it's unclear why it isn't functioning for you.

Strange, it wasn't running for me. I went and manually installed dprint with curl, then restarted everything and it worked!

> Consider adding this command to the `CONTRIBUTING.md` file.

Just added this.

I figured adding it before the `changesets` section made more sense so that `changesets` are generated with formatted code.

If you'd like me to swap it to after changesets just let me know.

Sorry for the fail 😔.

<img width="2056" alt="Screenshot 2025-05-13 at 7 49 45 AM" src="https://github.com/user-attachments/assets/d370cdae-0b3a-4435-a7a9-3d9d553e331c" />

DevChthonic commented:
Standing by with regards to the two test failures 🫡.

Wholly unfamiliar with Playwright.

I did not touch any other files except those two (CONTRIBUTING.md and cspell.jsonc).
